Report containing findings of both microtasks - NSFW classifier

HarshineeSriram opened the pull request.

HarshineeSriram committed:
Add files via upload

HarshineeSriram committed:
Merge pull request #1 from HarshineeSriram/HarshineeSriram-patch-1

Report for both microtasks - NSFW classifier

HarshineeSriram commented:
Test

abbasidaniyal commented:
Hey @HarshineeSriram this looks like a great start!
Just one thing I would like to point out. You have mentioned about the proposed implementation that you would like to carry forward. You can omit it from this report and keep it for the Final Application on the Outreachy website.

I will go over it in detail over the weekend and let you know what I feel!

HarshineeSriram committed:
Delete NSFW_Classifier_for_Wikimedia_Harshinee_Sriram.ipynb

HarshineeSriram committed:
Add files via upload

HarshineeSriram commented:
Hello @abbasidaniyal! Thank you so much for your response, I have omitted the last part.

abbasidaniyal reviewed (approved):
Hey! 
The detailed description of the classfiers is really nice. I think you've listed out most of the datasets out there. Great stuff!

Next thing you could try is to download a small portion of the datasets and do some EDA on it.  The will help in developing a better insight about the datasets. Let me know if you have something else in you mind!

HarshineeSriram commented:
Thank you so much for going through it @abbasidaniyal! Yes, I'll work on that right away.

HarshineeSriram committed:
Did some pre-processing to formulate a dataset

The next steps would be to explore the data, analyze it, and prepare it for training.

Link to the generated dataset: https://drive.google.com/file/d/1v-j1i28PFFQZDY5zm_mvJ71xSKGK-F2w/view?usp=sharing

HarshineeSriram commented:
Preprocessing to formulate a dataset.

HarshineeSriram committed:
Achieved an accuracy of 97.95% - validation set.

Attempted transfer learning on my curated dataset.

HarshineeSriram commented:
Hello @abbasidaniyal! I have added notebooks 2A) and 2B) where I experiment with Transfer Learning on the curated dataset. 2A contains the explanation/method of approach, and 2B contains the code.

abbasidaniyal commented:
Hey! I believe you can combine 1A and 1B into a single notebook. It's better to have the explanation and code in the same notebook, makes it easier to understand.
I like the approch of using lorem picsum inorder to build the SFW dataset. 
Just one thing, when you're splitting the dataset, you are using random.choice three times. So, it is quite possible the the train/val/test sets have duplicate data right?

Again, if you can combine 2A and 2B that'll be great. So this looks like a good first attempt. A couple of suggestions, you can use other callbacks like ReduceLROnPlateau, EarlyStopping to get better performance. If notice, the loss (train and val) are decreasing but not in a smooth way. Perhaps you could increase the number of trainable neurons or reduce the initial learning rate. However, it'll be better to iterate over these once the project begins.

In the "Next steps" section you have mentioned some short comings of the dataset. I personally would be more interesting in knowing your approch on how we can make it better.

abbasidaniyal commented:
Just one more things, inside the reports folder, create a new folder with you name or username. Use that folder to keep your files.
Eg: All my files should be in reports/abbasidaniyal/Dataset.ipynb
When we merge these PR's that will help in keeping everyone's reports sorted!

HarshineeSriram committed:
Merged 1A & 1B as well as 2A & 2B

HarshineeSriram committed:
Create .gitkeep

HarshineeSriram committed:
Transferred the main files to a username directory

HarshineeSriram committed:
Merge branch 'main' of https://github.com/HarshineeSriram/Wikimedia-NSFW-Classifier-Reports into main

HarshineeSriram commented:
Hello @abbasidaniyal ! I have merged notebooks 1A & 1B as well as 2A & 2B. I have also saved all the necessary files inside a new folder with my username @HarshineeSriram. 

About the dataset split, so once I had a folder with all possible images, I ran the duplicate checking script to remove images with different filenames but the same content. Also, every time I select a set of random images, I immediately move them to the new folder (before the next random selection) so that the main folder (which initially had all possible images) does not contain a copy. I could try running the duplicate checking script across the folders if you'd like.

As for additional callbacks, new neurons, and learning rate, I will work on them right away. Thank you so much for your suggestions. And, yes, I'll try solving some of these drawbacks soon and once I find good solutions, I'll post the reports :D

HarshineeSriram committed:
Applied supervisor's recommendations and improved my dataset.

HarshineeSriram commented:
Hi, @abbasidaniyal I have made the necessary hyper-parameter and callback changes as well as increased my data by 3 times the original size (with more variations).

HarshineeSriram committed:
Add files via upload

HarshineeSriram committed:
Add files via upload

HarshineeSriram committed:
Add files via upload

HarshineeSriram committed:
Update Outreachy Proposal.pdf

HarshineeSriram committed:
Delete Outreachy Proposal.pdf

abbasidaniyal commented:
Hi. This looks much more organized now. The loss in (4)/(3) is much better than what we had in (2) for InceptionV3!

Indeed MobileNet seems to give better results. You can try out MobileNetV2 as well. I think that can further improve the performance!

This looks like a great start. However, we should also test the model of images that are coming from a different data distribution. That will be better to understand the overall performace of the model.

PySceneDetect seems to be working well!

Note: From (3) you can remove the weights output. The file is over 10MB due to printing the weights.

HarshineeSriram committed:
Update 3) Hyperparameter and Dataset updates for notebook 2.ipynb

HarshineeSriram committed:
Add files via upload

HarshineeSriram committed:
Update 7) Implementing MobileNet_V2 model for transfer learning.ipynb

HarshineeSriram committed:
Add files via upload

HarshineeSriram committed:
Random image test

HarshineeSriram committed:
Moved notebook 8

HarshineeSriram committed:
Add files via upload

pep8speaks commented:
Hello @HarshineeSriram! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`reports/HarshineeSriram/keras-flask-image-classifier/application.py`](https://github.com/abbasidaniyal/Wikimedia-NSFW-Classifier-Reports/blob/40a73191942896936c27210b791595cfafab7767/reports/HarshineeSriram/keras-flask-image-classifier/application.py):

> [Line 2:80](https://github.com/abbasidaniyal/Wikimedia-NSFW-Classifier-Reports/blob/40a73191942896936c27210b791595cfafab7767/reports/HarshineeSriram/keras-flask-image-classifier/application.py#L2): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 3:80](https://github.com/abbasidaniyal/Wikimedia-NSFW-Classifier-Reports/blob/40a73191942896936c27210b791595cfafab7767/reports/HarshineeSriram/keras-flask-image-classifier/application.py#L3): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 8:38](https://github.com/abbasidaniyal/Wikimedia-NSFW-Classifier-Reports/blob/40a73191942896936c27210b791595cfafab7767/reports/HarshineeSriram/keras-flask-image-classifier/application.py#L8): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 12:1](https://github.com/abbasidaniyal/Wikimedia-NSFW-Classifier-Reports/blob/40a73191942896936c27210b791595cfafab7767/reports/HarshineeSriram/keras-flask-image-classifier/application.py#L12): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (3)
> [Line 15:80](https://github.com/abbasidaniyal/Wikimedia-NSFW-Classifier-Reports/blob/40a73191942896936c27210b791595cfafab7767/reports/HarshineeSriram/keras-flask-image-classifier/application.py#L15): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (135 > 79 characters)
> [Line 32:1](https://github.com/abbasidaniyal/Wikimedia-NSFW-Classifier-Reports/blob/40a73191942896936c27210b791595cfafab7767/reports/HarshineeSriram/keras-flask-image-classifier/application.py#L32): [E305](https://duckduckgo.com/?q=pep8%20E305) expected 2 blank lines after class or function definition, found 1
> [Line 45:1](https://github.com/abbasidaniyal/Wikimedia-NSFW-Classifier-Reports/blob/40a73191942896936c27210b791595cfafab7767/reports/HarshineeSriram/keras-flask-image-classifier/application.py#L45): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 59:1](https://github.com/abbasidaniyal/Wikimedia-NSFW-Classifier-Reports/blob/40a73191942896936c27210b791595cfafab7767/reports/HarshineeSriram/keras-flask-image-classifier/application.py#L59): [E305](https://duckduckgo.com/?q=pep8%20E305) expected 2 blank lines after class or function definition, found 1



##### Comment last updated at 2020-10-31 17:18:01 UTC

HarshineeSriram committed:
Add files via upload

HarshineeSriram committed:
Add files via upload

HarshineeSriram committed:
Delete README.md
